omnipaxos_core/src/storage.rs

```diff
@@ -190,12 +273,16 @@ where
     T: Entry,
     S: Snapshot<T>,
 {
-    pub(crate) fn with(storage: I) -> Self {
-        InternalStorage {
+    pub(crate) fn with(storage: I, batch_size: usize) -> Self {
+        let mut internal_store = InternalStorage {
             storage,
+            state_cache: StateCache::new(),
+            batch_size,
             _t: Default::default(),
             _i: Default::default(),
-        }
+        };
+        internal_store.load_cache();
```

Comment:
I realized that this might not work if the storage provided by the user is empty (i.e., when they create OmniPaxos for the first time). I think this could be detected if you try the test with `PersistentStorage`

To fix this, we should check if storage.get_promise() has the default ballot. If it is not, then we can load the cache.
```suggestion
        if storage.get_promised_round() != Ballot::default() {
                        internal_store.load_cache();
        }
```